src/GiraffeGenerator.Core/CodeGenNullChecking.fs

```diff
@@ -0,0 +1,212 @@
+﻿module CodeGenNullChecking
+
+open AST
+open ASTExt
+open FSharp.Compiler.SyntaxTree
+open OpenApi
+open OpenApiToAstTypesMatchingAndConversion
+
+type private PropertyPathSegment =
+  | Property of string
+  | NullableValue
+  | CollectionValue of PropertyPathSegment
+  | OptionValue of PropertyPathSegment
+
+let private isNullReference = "isNullReference"
+let private nullReferenceToOption = "nullReferenceToOption"
+let private defaultWithEmptySeq = Option.defaultValueExpr (longIdentExpr "Seq.empty")
+let private checkForUnexpectedNullsName = "checkForUnexpectedNulls"
+let generateNullCheckingHelpers () =
+    [
+        // isNullReference
+        letDecl false isNullReference ["v"] None // (=) doesn't work because F# thinks that C# can't fuck it up on nulls
+            ^ app (longIdentExpr "System.Object.ReferenceEquals")
+                (tupleComplexExpr [ identExpr "v"; SynExpr.Null(r) ])
+        // nullReferenceToOption
+        letDecl false nullReferenceToOption ["v"] None
```

Comment:
`Option.ofObj` maybe?

Reply:
same to `isNull` problem

Reply:
dropped null-checking as unstable and off-topic